core/src/types/masp.rs

```diff
@@ -15,23 +16,77 @@ use crate::types::string_encoding::{
     self, MASP_EXT_FULL_VIEWING_KEY_HRP, MASP_EXT_SPENDING_KEY_HRP,
     MASP_PAYMENT_ADDRESS_HRP,
 };
-use crate::types::token::MaspDenom;
+use crate::types::token::{Denomination, MaspDigitPos};
+
+/// The plain representation of a MASP aaset
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    Clone,
+    Debug,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+)]
+pub struct PreAssetType {
```

Comment:
I'm somehow not digging this name. Maybe something like `AssetData`?

Reply:
This sounds like a better name. Thanks!